Expanded installation section

Paulocracy opened the pull request:
I added an installation pathway for CPLEX in connection with CNApy, and tried to add the missing figure. As long as I didn't miss any simple way to do it, installing CPLEX seems to be quite tedious for a normal user :-| Should be automate this process somehow?

VonAlphaBisZulu commented on content/03.installation.md:
```diff
@@ -1,6 +1,61 @@
+
+### Additional solvers (optional)
+
+*Note: Additional solvers are only possible if you've installed CNApy according to the first main method, i.e. by using conda*
```
This should be possible in any case (installation via conda, via Windows installer and manually (i.e. cnapy-dev)). See below.

VonAlphaBisZulu commented on content/03.installation.md:
```diff
@@ -1,6 +1,61 @@
+
+#### Connect CPLEX to CNApy
+
+*Note: This connection is only possible if you've installed CNApy according to the first main method, i.e. by using conda*
```
This should be possible in any case (installation via conda, via Windows installer and manually (i.e. cnapy-dev)). See below.

VonAlphaBisZulu commented on content/03.installation.md:
````diff
@@ -1,6 +1,61 @@
+conda activate cnapy-dev
+```
+
+Once you're inside the cnapy-dev environment, install the CPLEX package by using pip as follows:
````
```suggestion
Once you activated the CNApy environment, navigate to your CPLEX installation (e.g. via ```cd "C:\Program Files\IBM\ILOG\CPLEX_Studio221"```) and install the CPLEX package as follows:
```

VonAlphaBisZulu commented on content/03.installation.md:
```diff
@@ -1,6 +1,61 @@
+
+*Note: This connection is only possible if you've installed CNApy according to the first main method, i.e. by using conda*
+
+Once installed CPLEX on your system (as described on [CPLEX's website](https://www.ibm.com/products/ilog-cplex-optimization-studio)), use conda in order to activate CNApy's conda environment:
```
```suggestion
Once installed CPLEX on your system (as described on [CPLEX's website](https://www.ibm.com/products/ilog-cplex-optimization-studio)), the API of your CPLEX installation must be connected to your CNApy's conda environment. Therefore, first open the command line tool and activate your CNApy's conda environment. Depending on your installation method, the name (or path) to your conda installation varies.
```

sthiele replied:
I would rephrase it like this: 
Once CPLEX is installed  on your system (as described on [CPLEX's website](https://www.ibm.com/products/ilog-cplex-optimization-studio)), the CPLEX Python API must be installed in the CNApy conda environment.

Also instead of: your CNApy's conda environment 
I would write: the CNApy conda environment

VonAlphaBisZulu commented on content/03.installation.md:
````diff
@@ -1,6 +1,61 @@
+```bash
+conda activate cnapy-dev
+```
+
````
```suggestion

or

```bash
conda activate cnapy-X.X.X```

or

```bash
conda activate C:\Users\YourUserName\cnapy```
```

VonAlphaBisZulu commented on content/03.installation.md:
````diff
@@ -1,6 +1,61 @@
+Once you're inside the cnapy-dev environment, install the CPLEX package by using pip as follows:
+
+```bash
+(cnapy-dev) pip install cplex
````
```suggestion
```
Correct me if I'm wrong, but this step and also the following steps should not be required, (apart from setup.py install)

Paulocracy replied:
I think they are necessary because cnapy-dev does not contain these packages by default, at least I had to install them :3

VonAlphaBisZulu replied:
There should be a way to only install dependencies. Maybe uninstall cplex community again with pip, but keep the dependencies?

Paulocracy replied:
The "cplex" package contains the CPLEX Community Edition but just as a "dummy" for the full version. Hence, uninstalling cplex wouldn't help. Instead, we have to tell cplex where the full CPLEX is.

VonAlphaBisZulu commented on content/03.installation.md:
````diff
@@ -1,6 +1,61 @@
+Then, install CPLEX's Python API by using the following command (for further details, see [this](https://www.ibm.com/docs/en/icos/20.1.0?topic=cplex-setting-up-python-api)):
+
+```bash
+(cnapy-dev) python setup.py install
````
```suggestion
(cnapy) python ./python/setup.py install
```

VonAlphaBisZulu commented on content/03.installation.md:
````diff
@@ -1,6 +1,61 @@
+(cnapy-dev) export PYTHONPATH=your_path_to_cplex/python/your_cplex_version/your_system
+```
+
+Under Windows, you can do this by searching for your system's "environmental variables" and adding the variable PYTHONNOUSERSITE with the value "your_path_to_cplex/python/your_cplex_version/your_system" using Windows's environmental variables setting window.
````
```suggestion
```
Also not sure if this is required...

Paulocracy replied:
This is actually required, because otherwise, the CPLEX Community Edition is used :-(

VonAlphaBisZulu replied:
Yes, but only because you installed the community edition via pip before. If you skip this step and only install the API from the full CPLEX installation you should be fine

VonAlphaBisZulu reviewed (changes requested):
Hey, I left you some comments. Not really sure, i

axelvonkamp commented:
For CPLEX (and Gurobi) it may perhaps be clearer to distinguish between the free demo version (which is OK for small models) that can be installed with pip without any further configuration and the full version which needs more configuration steps.

Paulocracy commented:
I just uploaded a commit which tries to merge all of your ideas, feel free to further comment it :-)
Does anyone of you have a successfull Gurobi connection to CNApy? Since I do not have it installed, I cannot deduce the installation instructions for it.

VonAlphaBisZulu commented:
Yes, gurobi works for me. The procedure is the same as for CPLEX. Gurobi's path to setup.py is in Windows:
C:\gurobi951\win64\setup.py

Again, there is a community version available from pip. People might be confused why Python uses the community version while they have the full Gurobi optimizer installed.

Paulocracy commented:
> Yes, gurobi works for me. The procedure is the same as for CPLEX. Gurobi's path to setup.py is in Windows: C:\gurobi951\win64\setup.py
> 
> Again, there is a community version available from pip. People might be confused why Python uses the community version while they have the full Gurobi optimizer installed.

Great :-) So how do we tell CNApy which Gurobi it should use? I.e., what is Gurobi's analogy to PYTHONPATH?

VonAlphaBisZulu commented:
I would say, the same as for CPLEX, just don't install the community version via pip. Or uninstall the community version (but not its dependencies) before installing the pro version. You'll never need the community and the full version installed at the same time in your environment anyways.

axelvonkamp commented:
Octave works with version 5.2 or higher. Also mention that the presence of an older octave version should be avaoided because this can mess with CNApy startup...

Paulocracy commented:
I updated the Octave explanation and added a first Gurobi installation explanation, however, since I'm not sure how to set up the "real" Gurobi instead of the demo version for optlang, I left a !!! which indicates that someone with knowledge about that should add the instructions :-)

Paulocracy commented:
Thanks to the new installation wizards in CNApy 1.0.9, the installation process is now much simpler and I uploaded an associated updating PR (#4). Therefore, all changes here are not neccessary anymore.
